lib/slack/web/api/endpoints/chat.rb

```diff
@@ -82,8 +82,14 @@ def chat_postMessage(options = {})
           def chat_update(options = {})
             throw ArgumentError.new('Required arguments :ts missing') if options[:ts].nil?
             throw ArgumentError.new('Required arguments :channel missing') if options[:channel].nil?
-            throw ArgumentError.new('Required arguments :text missing') if options[:text].nil?
+            throw ArgumentError.new('Required arguments :text or :attachments missing') if options[:text].nil? && options[:attachments].nil?
             options = options.merge(channel: channels_id(options)['channel']['id']) if options[:channel]
+            # attachments must be passed as an encoded JSON string
+            if options.key?(:attachments)
+              attachments = options[:attachments]
+              attachments = JSON.dump(attachments) unless attachments.is_a?(String)
+              options = options.merge(attachments: attachments)
+            end
```

Comment:
Oh we already do that in `chat_postMessage`, my bad. You can maybe make the above change in a future commit (but feel free to do it here too).